src/Functions/FunctionHelpers.cpp

```diff
@@ -70,8 +70,19 @@ ColumnsWithTypeAndName createBlockWithNestedColumns(const ColumnsWithTypeAndName
             }
             else if (const auto * const_column = checkAndGetColumn<ColumnConst>(*col.column))
             {
-                const auto & nested_col = checkAndGetColumn<ColumnNullable>(const_column->getDataColumn())->getNestedColumnPtr();
-                res.emplace_back(ColumnWithTypeAndName{ ColumnConst::create(nested_col, col.column->size()), nested_type, col.name});
+                const auto & nullable_column = checkAndGetColumn<ColumnNullable>(const_column->getDataColumn());
```

Comment:
It's very confusing to have const reference to const pointer here.

Suggest to change to: `const auto * nullable_column`

Reply:
Good catch, fixed, thanks